monkey/common/agent_plugins/agent_plugin_metadata.py

```diff
@@ -0,0 +1,48 @@
+from pathlib import PurePosixPath
+from typing import Any, Callable, Dict, Type, Union
+
+from pydantic import Field, validator
+
+from common.base_models import InfectionMonkeyBaseModel, InfectionMonkeyModelConfig
+
+from . import AgentPluginType, PluginName, PluginVersion
+
+
+class AgentPluginMetadata(InfectionMonkeyBaseModel):
+    """
+    Class for an Agent plugin's metadata
+
+    Attributes:
+        :param name: Plugin name
+        :param type: Plugin type
+        :param resource_path: Path of the plugin file in the AWS S3 bucket
```

Comment:
This data structure shouldn't make assumptions about what service is used to serve it.